Fix compilation failure in `datafusion-ext-plans` after primitive sort fast path change

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->

The current `master` branch fails to compile `datafusion-ext-plans`:

```text
error[E0624]: method `len` is private
  --> native-engine/datafusion-ext-plans/src/sort_exec.rs
   |
   | create_zero_column_batch(sorted_indices.len())
```
The primitive sort fast path changed sorted_indices from `Vec<u32>` to `Option<Vec<u32>>`. The zero-column batch path still calls `len()` as if it were a `Vec`.

This causes CI failures for PRs based on the current master.

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->

`cargo check -p datafusion-ext-plans --lib`

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

compile successfully.

**Screenshots**
<!--
If applicable, add screenshots to help explain your problem.
-->

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/native-engine/datafusion-ext-plans/src/sort_exec.rs
+++ b/native-engine/datafusion-ext-plans/src/sort_exec.rs
@@ -722,7 +722,7 @@ impl ExternalSorter {
                 take_batch(batch, sorted_indices.expect("non-fast-path indices"))?
             }
         } else {
-            create_zero_column_batch(sorted_indices.len())
+            create_zero_column_batch(sorted_indices.as_ref().map_or(keys.num_rows(), Vec::len))
         };
 
         // add to in-mem blocks

